liblangutil/ParserBase.h

```diff
@@ -38,6 +39,7 @@ class ParserBase
 	explicit ParserBase(ErrorReporter& errorReporter): m_errorReporter(errorReporter) {}
 
 	std::shared_ptr<std::string const> const& sourceName() const;
+	CharStream const* charStream() const { return &m_scanner->charStream(); }
```

Comment:
Why not return by reference?

Reply:
Sorry, I don't remember the conclusion of our discussion here. Why is it fine to return by pointer here but return by reference in the scanner?

Reply:
It's returning a `shared_ptr<CharStream>` now. I wonder why Github isn't showing it here.